tests/OK/292.lp

```diff
@@ -0,0 +1,21 @@
+set flag "print_domains" on
+
+symbol R : TYPE
+constant symbol zero : R
+symbol d2_1 : (R ⇒ R ⇒ R) ⇒ (R ⇒ R ⇒ R)
+
+// The following are semantically equivalent
+rule d2_1 (λ_ y, &F[y]) → λ_ _,zero // warning
+rule d2_1 (λ_ y, &_[y]) → λ_ _,zero // warning
+rule d2_1 (λ_ y, _    ) → λ_ _,zero // no warning
+
+// The following two are NOT semantically the same
+rule d2_1 (λx y, &_[y]) → λ_ _,zero // no warning
+rule d2_1 (λ_ y, _    ) → λ_ _,zero // no warning
+// The difference is that in the former variable "x" may appear in the type
+// of the "y" variable. That is not the case in the second line.
+
```

Comment:
and is `λ_x y, _` equivalent to `λ_ y, _`?

Reply:
No, it would be equivalent to `λx y, _` in the current implementation.

Reply:
This is not the semantics I expect.

Reply:
I have mixed feelings about that. This is the semantics in most programming languages I know (including OCaml, where you can write `fun _x -> _x`, so clearly identifiers starting with an underscore are in scope).

Reply:
I don't like this. I prefer https://github.com/Deducteam/lambdapi/pull/293#discussion_r351349302.

Reply:
I don't particularly like it either, but taking different conventions as in every other language might not be the greatest idea.

Reply:
At least, could you please change your comment? The notion of semantical equivalence is undefined. And all these rules should really be equivalent. So, it is better to say so and open an issue for fixing the subject-reduction algorithm.

Reply:
Done.

Reply:
I opened https://github.com/Deducteam/lambdapi/issues/294 .

Reply:
I prefer to write:
```
// The following rules are equivalent
rule d2_1 (λ_ y, &F[y]) → λ_ _,zero // warning
rule d2_1 (λ_ y, &_[y]) → λ_ _,zero // warning
rule d2_1 (λ_ y, _    ) → λ_ _,zero // no warning
rule d2_1 (λx y, &_[y]) → λ_ _,zero // no warning
```
and not mention than there is currently a difference between the last two rules because this is a bug that will be fixed later, and we may forget to remove this comment when the problem will be fixed.

Reply:
I just say that they are not the same and exactly explain that. I'm not gonna say that the last one is equivalent because that is simply not true. I can however add a reference to the issue.